scrapy/utils/deprecate.py

```diff
@@ -126,9 +126,7 @@ def _clspath(cls, forced=None):
     return f'{cls.__module__}.{cls.__name__}'
 
 
-DEPRECATION_RULES = [
-    ('scrapy.telnet.', 'scrapy.extensions.telnet.'),
-]
+DEPRECATION_RULES = []
 
 
 def update_classpath(path):
```

Comment:
I think may we should remove this function as well, since now it will just return the provided path, make sense @wRAR?

And probably pass a lambda function to `convert` parameter below (and remove from tests).
https://github.com/scrapy/scrapy/blob/45c2bd7d9c1d6adf0e051842c422c5d6af1d0a17/scrapy/utils/conf.py#L15

Reply:
If there is a possibility of repopulating the `DEPRECATION_RULES` list in the future we should keep the `update_classpath` method otherwise similar logic will need to be re-written.

Reply:
I think it will be not the case since `@deprecated` decorator already have this functionality:
https://github.com/scrapy/scrapy/blob/45c2bd7d9c1d6adf0e051842c422c5d6af1d0a17/scrapy/utils/decorators.py#L9

But let's see what Scrapy team has to say about.

Reply:
`update_classpath` is used in the code that parses object dicts such as `DOWNLOADER_MIDDLEWARES`: https://github.com/scrapy/scrapy/blob/45c2bd7d9c1d6adf0e051842c422c5d6af1d0a17/scrapy/utils/conf.py#L15

I think you can't use `@deprecated` for this.

Reply:
Yeah but the logic depends all on `DEPRECATION_RULES` variable right?

```python
    for prefix, replacement in DEPRECATION_RULES:
        if isinstance(path, str) and path.startswith(prefix):
            new_path = path.replace(prefix, replacement, 1)
            warnings.warn(f"`{path}` class is deprecated, use `{new_path}` instead",
                          ScrapyDeprecationWarning)
            return new_path
    return path
```
If nothing is provided it will just return what was originally send by parameter right?

Reply:
Right.